jxl/src/frame.rs

```diff
@@ -190,6 +256,27 @@ impl Frame {
 
         Ok(())
     }
+
+    pub fn finalize(mut self) -> Result<Option<DecoderState>> {
+        if self.header.can_be_referenced {
+            // TODO(firsching): actually decode the image here, instead of setting it
+            // to a blank image here, once we can decode images.
+            self.decoder_state.reference_frames[self.header.save_as_reference as usize] =
+                Some(ReferenceFrame::blank(
+                    self.header.width as usize,
+                    self.header.height as usize,
+                    // Set num_channels to "1" here unconditionally for now, since we will use this
+                    // channel for the size checks
+                    1,
```

Comment:
why not 3 + self.decoder_state.extra_channel_info().len()?

Reply:
sure, why not, will be deleted anyhow. (I'm testing on a grayscale image, so "1" seemed more natural)